mito-ai/mito_ai/file_uploads/handlers.py

```diff
@@ -0,0 +1,120 @@
+        self, filename: str, file_data: bytes, chunk_number: str, total_chunks: str
+    ) -> None:
+        """Handle chunked file upload."""
+        chunk_num = int(chunk_number)
+        total_chunks_num = int(total_chunks)
+
+        # Save chunk to temporary file
+        self._save_chunk(filename, file_data, chunk_num)
+
+        # Check if all chunks are received and reconstruct if complete
+        if self._are_all_chunks_received(filename, total_chunks_num):
+            self._reconstruct_file(filename, total_chunks_num)
+            self._send_chunk_complete_response(filename)
+        else:
+            self._send_chunk_received_response(chunk_num, total_chunks_num)
+
+    def _handle_regular_upload(self, filename: str, file_data: bytes) -> None:
+        """Handle regular (non-chunked) file upload."""
+        with open(filename, "wb") as f:
+            f.write(file_data)
+
+        self.write({"success": True, "filename": filename, "path": filename})
+
+    def _save_chunk(self, filename: str, file_data: bytes, chunk_number: int) -> None:
+        """Save a chunk to a temporary file."""
+        chunk_filename = f"{filename}.part{chunk_number}"
+        with open(chunk_filename, "wb") as f:
+            f.write(file_data)
+
+    def _are_all_chunks_received(self, filename: str, total_chunks: int) -> bool:
+        """Check if all chunks for a file have been received."""
+        for i in range(1, total_chunks + 1):
+            if not os.path.exists(f"{filename}.part{i}"):
+                return False
+        return True
+
+    def _reconstruct_file(self, filename: str, total_chunks: int) -> None:
+        """Reconstruct the final file from all chunks and clean up temporary files."""
+        with open(filename, "wb") as final_file:
+            for i in range(1, total_chunks + 1):
+                chunk_filename = f"{filename}.part{i}"
+                with open(chunk_filename, "rb") as chunk_file:
+                    final_file.write(chunk_file.read())
+                # Clean up chunk file
+                os.remove(chunk_filename)
+
+    def _send_chunk_complete_response(self, filename: str) -> None:
+        """Send response indicating all chunks have been processed and file is complete."""
+        self.write(
+            {
+                "success": True,
+                "filename": filename,
+                "path": filename,
+                "chunk_complete": True,
+            }
+        )
+
+    def _send_chunk_received_response(
+        self, chunk_number: int, total_chunks: int
+    ) -> None:
+        """Send response indicating a chunk was received but file is not yet complete."""
+        self.write(
+            {
+                "success": True,
+                "chunk_received": True,
+                "chunk_number": chunk_number,
+                "total_chunks": total_chunks,
+            }
+        )
+
+    def _handle_error(self, error_message: str, status_code: int = 500) -> None:
```

Comment:
When this raises an error, we don't share any information with the user. Can we do the following: 

1. Use the Juptyer Notification system to display the error message
2. Log the error so we can figure out how to improve the feature!